### PR TITLE
[BUG] Unequal length bounding matrix fixes

### PR DESCRIPTION
**Pull request by chrisholder**
#### Reference Issues/PRs
Fixes issue #492 

#### What does this implement/fix? Explain your changes.
This PR fixes a bug with the bounding matrix where if the second time series was bigger than the first the sakoe chiba window would not be created properly.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [x] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).

##### For developers with write access
- [x] Optionally, I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Comment by CodeLionX** [comment]
You could add a test for `create_bounding_matrix` with the different input lengths (for `x_size` and `y_size`) to catch this bug in the future and maybe also for other bounding matrices.

**Comment by chrisholder** [comment]
> You could add a test for `create_bounding_matrix` with the different input lengths (for `x_size` and `y_size`) to catch this bug in the future and maybe also for other bounding matrices.

Did this and found a bug in the process!

**Comment by chrisholder** [comment]
I have also raised a issue for Itakura parallelogram to support unequal length in the future. See issues #1300.

File: aeon/distances/_bounding_matrix.py [hunk 1]
```diff
@@ -34,14 +35,14 @@ def create_bounding_matrix(
...
     --------
     >>> create_bounding_matrix(8, 8, window=0.5)
-    array([[ True,  True,  True,  True,  True, False, False, False],
-           [ True,  True,  True,  True,  True,  True, False, False],
-           [ True,  True,  True,  True,  True,  True,  True, False],
-           [ True,  True,  True,  True,  True,  True,  True,  True],
-           [ True,  True,  True,  True,  True,  True,  True,  True],
-           [False,  True,  True,  True,  True,  True,  True,  True],
-           [False, False,  True,  True,  True,  True,  True,  True],
-           [False, False, False,  True,  True,  True,  True,  True]])
```
Comment (chrisholder) [inline aeon/distances/_bounding_matrix.py:44]: Fixed this issue were bounding matrix was 1 to big. The fixed version is now equivalent to tslearn (I tested over 50 different length over 10 different window sizes (0.1, 0.2, 0.3,...0.9. 1.0)

**Review by TonyBagnall — APPROVED** [review]
LGTM, nice one

**Pull request merged**